rclpy/rclpy/impl/rmw_implementation_tools.py

```diff
@@ -85,23 +83,8 @@ def import_rmw_implementation():
     if __rmw_implementation_module is not None:
         return __rmw_implementation_module
     available_implementations = get_rmw_implementations()
-    if __selected_rmw_implementation is None:
-        logger = logging.getLogger('rclpy')
-        # prefer FastRTPS, otherwise first in alphabetical order
-        # the same logic is implemented in
-        # rmw_implementation_cmake/cmake/get_default_rmw_implementation.cmake
-        default_impl = 'rmw_fastrtps_cpp'
-        if default_impl in available_implementations:
-            select_rmw_implementation(default_impl)
-        elif available_implementations:
-            select_rmw_implementation(available_implementations[0])  # select the first one
-        else:
-            raise NoImplementationAvailableException()
-        logger.debug("Implicitly selecting the '{0}' rmw implementation."
-                     .format(__selected_rmw_implementation))
-    module_name = '._rclpy__{rmw_implementation}'.format(
-        rmw_implementation=__selected_rmw_implementation,
-    )
+    __selected_rmw_implementation = 'rmw_implementation'
```

Comment:
> As mentioned above I have already removed these in this PR in 9e05978.

I know I said that 😛:

> > remove the code and tests for the Python selection of rmw impl (as is proposed in this pr)

My point was that if someone else reviews and agrees with me then you'll need to do the other thing.

> Again I don't see the point of squeezing as much changes as possible into this PR set. 

It's not about squeezing as much as possible in, it is about making sure the master branch stays in the best state possible even between incremental pr's. When someone asks me to review a pr, I look for anything that smells bad like: failing test, failing style, logical inconsistency, dead code, broken features. I think it is acceptable to table any of those things by marking them with a TODO or tracking them in a ticket, but only if there is no plan to fix them in the immediate future. If you're just going to merge this pr and then do those things right afterwards, then I don't see much benefit to splitting them up. The only other reason to merge short of the goal like this is to prevent getting out of sync with others, but since you're the only one working in this part of the code right now, that's not an issue in my opinion. So from my perspective you're neither at risk of conflicts nor are you going to defer doing these things for very long and so I think you should do them in these current pr's.

I can't see a good reason to do it in two steps. In contrast, you may discover something important when removing these functions (like that they're still being used somewhere that you missed) which would be important to find and fix before putting it to master.

I appreciate that you're trying to keep the changes small and reviewable, but I just think they should also be cohesive and that splitting the pr's short of removing now vestigial functions just doesn't make sense.

> Imo this discussion is slowing the progress more down then if this would have been merged. Then the remaining tasks could have already been started / done by now.

Imo if you just removed them and restarted CI when I mentioned it, then we'd also be done by now. The size of the pr doesn't affect the time it takes to run CI or anything. In fact the time it would take you to make new branches and pr's for the additional changes might make it take longer.

> I can remove the CMake functions in the corresponding PR if that makes a difference for you.

I think it is worthwhile to do that at the same time as these others changes, yes. I don't know what the rest of the @ros2/team is doing today (hint, hint -- nudge, nudge 😄), but they could override me and that would be fine.